letsencrypt/client/display/revocation.py

```diff
@@ -0,0 +1,95 @@
+"""Revocation UI class."""
+import os
+
+import zope.component
+
+from letsencrypt.client import interfaces
+from letsencrypt.client.display import util as display_util
+
+util = zope.component.getUtility  # pylint: disable=invalid-name
+
+
+def choose_certs(certs):
+    """Choose a certificate from a menu.
+
+    :param list certs: List of cert dicts.
+
+    :returns: selection (zero-based index)
+    :rtype: int
+
+    """
+    while True:
+        code, selection = _display_certs(certs)
+        if code == display_util.OK:
+            return selection
+        elif code == display_util.HELP:
+            more_info_cert(certs[selection])
+        else:
+            exit(0)
```

Comment:
Thanks for pointing this out.  After the refactor, I didn't realize that I was infinite looping in two separate places... both revoke_from_menu and choose_certs.  I can probably refactor this exit out of there if I add some return logic in revoke_from_menu.